lnd.go

```diff
@@ -1620,108 +1563,144 @@ func initializeDatabases(ctx context.Context,
 
 	startOpenTime := time.Now()
 
-	databaseBackends, err := cfg.DB.GetBackends(ctx, cfg.localDatabaseDir())
+	databaseBackends, err := cfg.DB.GetBackends(
+		ctx, cfg.graphDatabaseDir(), cfg.networkDir, filepath.Join(
+			cfg.Watchtower.TowerDir,
+			cfg.registeredChains.PrimaryChain().String(),
+			lncfg.NormalizeNetwork(cfg.ActiveNetParams.Name),
+		), cfg.WtClient.Active, cfg.Watchtower.Active,
+	)
 	if err != nil {
-		return nil, nil, nil, fmt.Errorf("unable to obtain database "+
+		return nil, nil, fmt.Errorf("unable to obtain database "+
 			"backends: %v", err)
 	}
 
-	// If the remoteDB is nil, then we'll just open a local DB as normal,
-	// having the remote and local pointer be the exact same instance.
+	// With the full remote mode we made sure both the graph and channel
+	// state DB point to the same local or remote DB and the same namespace
+	// within that DB.
+	const (
+		lblGraphDB    = "graph"
+		lblStateDB    = "channel state"
+		lblHintDB     = "height hint"
+		lblMacDB      = "macaroon"
+		lblSphinxDB   = "sphinxreplay"
+		lblWtClientDB = "watchtower client"
+		lblWtServerDB = "watchtower server"
+	)
 	var (
-		localChanDB, remoteChanDB *channeldb.DB
-		closeFuncs                []func()
+		dbs = &databaseInstances{
+			heightHintDB: databaseBackends.HeightHintDB,
+			macaroonDB:   databaseBackends.MacaroonDB,
+			decayedLogDB: databaseBackends.DecayedLogDB,
+			walletDB:     databaseBackends.WalletDB,
+		}
+		closeFuncs = map[string]func() error{
+			// We use the non-decorated kvdb based close functions
+			// here in case we need to clean up early because of an
+			// error during initialization. We'll replace some of
+			// these with the decorated channeldb close functions
+			// once we've set them up.
+			lblGraphDB:  databaseBackends.GraphDB.Close,
+			lblStateDB:  databaseBackends.ChanStateDB.Close,
+			lblHintDB:   databaseBackends.HeightHintDB.Close,
+			lblMacDB:    databaseBackends.MacaroonDB.Close,
+			lblSphinxDB: databaseBackends.DecayedLogDB.Close,
+		}
```

Comment:
Don't you want to return the close funcs from `GetBackends` to not repeat yourself?

Reply:
Yeah, makes a lot of sense. Was able to simplify this quite a bit. What do you think?